pandas/tests/util/test_show_versions.py

```diff
@@ -26,13 +33,56 @@
     "ignore:Distutils:UserWarning"
 )
 @pytest.mark.filterwarnings("ignore:Setuptools is replacing distutils:UserWarning")
-def test_show_versions(capsys):
+def test_show_versions(tmpdir):
+    as_json = os.path.join(tmpdir, "test_output.json")
+
+    pd.show_versions(as_json=as_json)
+
+    # make sure that the file was created
+    assert os.path.exists(as_json)
+
+    with open(as_json) as fd:
+        contents = fd.readlines()
+        str_contents = "".join(contents)
+
+        # make sure that there was output to the file
+        assert str_contents
+
+        # check if file output is valid JSON, will raise an exception if not
+        dict_check = json.loads(str_contents)
```

Comment:
any reason to do this over
```python
    with open(as_json) as fd:
        dict_check = json.load(fd)
```
?

Also, I think the naming is a bit confusing, maybe rename `dict_check` to `result` and `version_elements` to `expected`

Reply:
None that I can think of. It seems I made this more complicated than it need to be. Thanks !